File: Step 3/c.cs
using System;

class Program
{
    static void Main(string[] args)
    {
        string[] dimensions = Console.ReadLine().Split(' ');
        int n = int.Parse(dimensions[0]);
        int m = int.Parse(dimensions[1]);
        string[] candlePosition = Console.ReadLine().Split(' ');
        int x = int.Parse(candlePosition[0]);
        int y = int.Parse(candlePosition[1]);
        double result = AreaDifference(n, m, x, y);
        Console.WriteLine($"{result:F3}");
    }

    static double AreaDifference(int n, int m, int x, int y)
    {
        double totalArea = n * m;
        double part1_1 = (x * y) / 2.0;
        double part1_2 = totalArea - part1_1;
        double diff1 = Math.Abs(part1_1 - part1_2);
        double part2_1 = ((n - x) * y) / 2.0;
        double part2_2 = totalArea - part2_1;
        double diff2 = Math.Abs(part2_1 - part2_2);
        double part3_1 = ((n - x) * (m - y)) / 2.0;
        double part3_2 = totalArea - part3_1;
        double diff3 = Math.Abs(part3_1 - part3_2);
        double part4_1 = (x * (m - y)) / 2.0;
        double part4_2 = totalArea - part4_1;
        double diff4 = Math.Abs(part4_1 - part4_2);
        return Math.Min(Math.Min(diff1, diff2), Math.Min(diff3, diff4));
    }
}

File: Step 3/c.py
def area_difference(n1, n2, n3, n4):
    total_area = n1 * n2
    part1_1 = (n3 * n4) / 2
    part1_2 = total_area - part1_1
    diff1 = abs(part1_1 - part1_2)
    part2_1 = ((n1 - n3) * n4) / 2
    part2_2 = total_area - part2_1
    diff2 = abs(part2_1 - part2_2)
    part3_1 = ((n1 - n3) * (n2 - n4)) / 2
    part3_2 = total_area - part3_1
    diff3 = abs(part3_1 - part3_2)


    part4_1 = (n3 * (n2 - n4)) / 2
    part4_2 = total_area - part4_1
    diff4 = abs(part4_1 - part4_2)

    return min(diff1, diff2, diff3, diff4)

num1, num2 = map(int, input().split())
num3, num4 = map(int, input().split())

result = area_difference(num1, num2, num3, num4)

print(f"{result:.3f}")
